home/management/commands/crawlposts.py

```diff
@@ -113,7 +113,10 @@ def cleantitle(title):
 
 def get_or_create_post(blog, title, link, date):
     try:
-        post = Post.objects.get(blog=blog, title=title)
+        # The previous code checked only for url, and therefore, the db can
+        # have posts with duplicate titles. So, we check if there is atleast
+        # one post with the title -- using `filter.latest` instead of `get`.
+        post = Post.objects.filter(blog=blog, title=title).latest('date_updated')
```

Comment:
Great! I was worried this wouldn't work since duplicate posts will likely have identical timestamps (the date the post says it was created, not the datetime blaggregator crawled it). But Django picks on if there's a tie. Great!